File: core/src/main/kotlin/com/example/toyracers/track/TrackId.kt
package com.example.toyracers.track

/** Stable identifiers and player-facing names for built-in tracks. */
enum class TrackId(
    val value: String,
    val displayName: String,
) {
    LIVING_ROOM("track-01", "LIVING ROOM"),
    BATHROOM("track-02", "BATHROOM"),
    ;

    companion object {
        fun fromValue(value: String): TrackId =
            entries.firstOrNull { it.value == value }
                ?: throw IllegalArgumentException("Unknown track: $value")
    }
}

File: dart/lib/simulation/track/track_id.dart
/// Stable IDs and display names of the built-in track definitions.
enum TrackId {
  livingRoom(id: 'track-01', displayName: 'LIVING ROOM'),
  bathroom(id: 'track-02', displayName: 'BATHROOM');

  const TrackId({required this.id, required this.displayName});

  final String id;
  final String displayName;

  static TrackId fromId(String id) {
    for (final trackId in TrackId.values) {
      if (trackId.id == id) {
        return trackId;
      }
    }
    throw ArgumentError.value(id, 'id', 'Unknown track');
  }
}
